### PR TITLE
lintpkgforfile() while loop replaced with for loop to fix #170

### LINKED ISSUE #170: The linter is hanging
Running Julia 0.5 on Windows 10, `lintfile` hangs on a `.jl` file with the following contents:

```julia
y = x +1
0 = 1
```

### PR DESCRIPTION
**Pull request by TeroFrondelius**

### CODE REVIEW

**Comment by TeroFrondelius** [comment]
I will add one test.

File: src/Lint.jl [hunk 1]
```diff
@@ -70,7 +70,12 @@ If file is in base lint all files in base dir.
 function lintpkgforfile{T<:AbstractString}(path::T, ctx::LintContext=LintContext())
     path = abspath(path)
     if ispath(ctx.path)
-        while path != "/"
+        if is_windows()
+            len = length(matchall(r"\\", path))+1
```
Comment (TotalVerb) [inline src/Lint.jl]: `count(x -> x == '\\', path) + 1`

File: src/Lint.jl [hunk 2]
```diff
@@ -70,7 +70,12 @@ If file is in base lint all files in base dir.
...
+        else
+            len = length(matchall(r"/", path))
```
Comment (TotalVerb) [inline src/Lint.jl]: `count(x -> x == '/', path)`

File: src/Lint.jl (hunk 1 above)
Comment (TeroFrondelius) [inline src/Lint.jl]: I removed "+1" and tests work. I think we can removed it. I think I made a mistake yesterday with +1.

**Comment by TeroFrondelius** [comment]
Hi @Michael-Klassen, is there some needs for change or is this ok to merge?

**Pull request merged**